arch-design/multicloud/Multi_Cloud_HA.md

```diff
@@ -0,0 +1,104 @@
+There are two types of downtimes- scheduled and unscheduled. A scheduled downtime is a result of maintenance, which is unavoidable. This includes applying patches, updating softwares or even changes in the database schema. An unscheduled downtime is, however, caused by some unforeseen event, like hardware or software failure. This can happen due to power outages or failure of a component. Scheduled downtimes are generally excluded from performance calculations.
+The prime objective of implementing High Availability architecture is to make sure your system or application is configured to handle different loads and different failures with minimal or no downtime. There are multiple components that help you in achieving this, and we will be discussing them briefly.
+
+## Goal
+The document provides the use case, design and implementation guidelines for supporting High Availability in SODA Multi Cloud project. It proposes a generic method of providing High Availability for the Multi Cloud project to begin with and include the other projects in the subsequent releases. To begin with, Kubernetes is considered to provide the High availability for the containers of the Multi Cloud service. 
+
+
+## Motivation and background
+As SODA provides a critical data management platform for both On-Prem and for Cloud based workloads, it is imperative that there is no bottleneck and the data and services be available even in case of any node going down. However, the current  architecture where all the components of SODA installed on a single node are a Single Point of Failure. In case of any failure of any of the services or the node going down, the data will not be accessible to the user. Hence, it's recommended to have the SODA installation in a High Available configuration, which can withstand failure of nodes and the data and services are accessible in case of any node or service going down. 
+
+## Non-Goals
+ - Clients access to the data and services is seamless. 
+ - The Services should scale up / down to provide data and services to users.
+ - Installation / Uninstallation of SODA provides options to install SODA Multi Cloud in standalone / HA mode. 
+
+
+## Assumptions and Constraints
+TBD
+
+## Use case View
+- High Availability : To provide High Availability to the SODA Multi Cloud Deployment.
+Scale up / down as per the need of data movement. 
+
+
+
+![Multi Cloud HA Design](resources/multicloud_ha.PNG)
+                        
+
+Here is the High Level Architecture for providing HA for SODA Multi Cloud services.  
+
+
+As seen in Figure-1, the scope of I release is to provide  High Availability for the SODA Multi Cloud Services to begin with and gradually migrate the other components over the subsequent releases. 
+With this in mind, the SODA node can serve as the  Container Orchestrator (K8s / Mesos / Docker Swarm) Master node and the Multi Cloud services be deployed on the Deployment Cluster as shown in Figure - 1.
```

Comment:
Why not we consider just HA tools like HAProxy for this usecase, Do we really need an orchestrator to deploy ?

Reply:
This analysis document aims to provide High Availability for Multi Cloud Services (which is a microservices based containerized solution) . HA Proxy only provides HA - LB for nodes, it does not provide against Service failures and Service scale. Hence a deployment solution which provides High Availability in case of a Node Failure , Service Failure , Cluster / Region failure is the right choice to go with. As it is a one time decision, its recommended that such solutions be chosen which can be extended as needed to mitigate against any of the Node / Service / DC failures.